docs/source/configuration/events.mdx

```diff
@@ -0,0 +1,195 @@
+---
+title: Events
+subtitle: Log or trace events in the router's request lifecycle
+description:
+---
+
+Events are used to signal when something of note happens in the [Apollo Router's request lifecycle (pipeline)](../customizations/overview/#the-request-lifecycle). Events are output to both logs and traces.
+
+The router's pipeline has three major services:
+
+* **Router service** - Handles an incoming request before it is parsed. Works within a context of opaque bytes.
+* **Supergraph service** - Handles a request after it has been parsed and before it is sent to the subgraph. Works within a GraphQL context.
+* **Subgraph service** - Handles a request after it has been sent to the subgraph. Works within a  GraphQL context.
+
+You can configure in `router.yaml` events for each service. Events can be standard or custom, and they can be triggered by configurable conditions.
+
+For example, the router service can be configured with standard events (`request`, `response`, `error`), and a custom event (`my.event`) with a condition:
+
+```yaml title="router.yaml"
+telemetry:
+  events:
+    router:
+      # Standard events
+      request: info
+      response: info
+      error: info
+
+      # Custom events
+      my.event:
+        message: "my event message"
+        level: info
+        on: request
+        attributes:
+          http.response.body.size: false
+        # Only log when the x-log-request header is `log` 
+        condition:
+          eq:
+            - "log"
+            - request_header: "x-log-request"
+        
+    supergraph:
+        # ...
+    subgraph:
+        # ...
+```
+
+### Event configuration
+
+#### `router`|`supergraph`|`subgraph`
+
+The `router`, `supergraph` and `subgraph` sections are used to define custom event configuration for the `router`, `supergraph` and `subgraph` services of the router, respectively.
+
+```yaml title="router.yaml"
+telemetry:
+  events:
+    router:
+      # ...     
+    supergraph:
+      # ...
+    subgraph:
+      # ...
+```
+
+Each service has a set of standard events that can be configured:
+
+* `request` - The request has been received.
+* `response` - The response has been sent.
+* `error` - An error in the router pipeline has occurred.
+
+<Note>
+
+The `error` level applies only to router pipeline errors, not GraphQL errors.
+
+</Note>
+
+To configure these events, set the level to `trace`, `info`, `warn`, `error` or `off` (default).
```

Comment:
What about `debug` level?

(This applies to other locations below where it has been left out)

Reply:
[b55cc92](https://github.com/apollographql/router/pull/4129/commits/b55cc9241e5f45b0f630f698eacf7f39c84d4ae2)